Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs

```diff
@@ -24,16 +24,39 @@ protected bool hasDog
 		}
 	}
 
+	protected Dog dog
+	{
+		get
+		{
+			return _dog;
+		}
+		set
+		{
+			// Fixes ref on previous dog
+			if(_dog != null)
+			{
+				_dog.LeaveCurrentSlot();
+			}
+			// Assigns slot to new dog (assuming the new value is not null)
+			if(value != null)
+			{
+				value.AssignSlot(this);
+			}
+			_dog = value;
+		}
+	}
+
 	bool hasButton {
 		get {
 			return button != null;
 		}
 	}
 		
     protected DogDescriptor dogInfo;
-	protected Dog dog;
+
     protected Image[] images;
 
+	Dog _dog;
```

Comment:
They should always be using the mutator (ensures that the Dog ref is up to date)